nautobot/ipam/tests/test_models.py

```diff
@@ -962,6 +962,93 @@ def setUp(self):
+        prefix_status = Status.objects.get_for_model(Prefix).first()
+
+        parent = Prefix.objects.create(prefix="10.1.1.0/24", namespace=namespace, status=prefix_status)
+        default_namespace_parent = Prefix.objects.create(
+            prefix="10.0.0.0/24", namespace=default_namespace, status=prefix_status
+        )
+
+        ipaddress = IPAddress.objects.create(address="10.1.1.1/24", namespace=namespace, status=ipaddr_status)
+        default_namespace_ipaddress = IPAddress.objects.create(
+            address="10.0.0.1/24", namespace=default_namespace, status=ipaddr_status
+        )
+        mask_length = 24
+
+        with self.subTest("Assert retrieve"):
+            ip_obj, created = IPAddress.objects.get_or_create(
+                host=ipaddress.host,
+                mask_length=mask_length,
+                namespace=namespace,
+                status=ipaddr_status,
+            )
+            self.assertEqual(ip_obj, ipaddress)
+            self.assertFalse(created)
+
+        with self.subTest(
+            "Assert get_or_create utilizes default namespace when retrieving parent if no namespace is provided"
+        ):
+            ip_obj, created = IPAddress.objects.get_or_create(
+                host=default_namespace_ipaddress.host,
+                mask_length=default_namespace_ipaddress.mask_length,
+                status=ipaddr_status,
+            )
+            self.assertEqual(ip_obj, default_namespace_ipaddress)
+            self.assertFalse(created)
+
+        with self.subTest("Assert create"):
+            new_host = "10.0.0.2"
+            ip_obj, created = IPAddress.objects.get_or_create(
+                host=new_host,
+                mask_length=mask_length,
+                status=ipaddr_status,
+            )
+            self.assertEqual(ip_obj.host, new_host)
+            self.assertEqual(ip_obj.mask_length, mask_length)
+            self.assertEqual(ip_obj.parent, default_namespace_parent)
+            self.assertEqual(ip_obj.parent.namespace, default_namespace)
+            self.assertTrue(created)
+
+        with self.subTest("Assert create explicitly defining a non default namespace"):
+            new_host = "10.1.1.2"
+            ip_obj, created = IPAddress.objects.get_or_create(
+                host=new_host, mask_length=mask_length, status=ipaddr_status, namespace=namespace
+            )
+            self.assertEqual(ip_obj.host, new_host)
+            self.assertEqual(ip_obj.mask_length, mask_length)
+            self.assertEqual(ip_obj.parent, parent)
+            self.assertEqual(ip_obj.parent.namespace, namespace)
+            self.assertTrue(created)
+
+        with self.subTest("Assert if host is not present"):
+            ip_obj, created = IPAddress.objects.get_or_create(host=ipaddress.host, status=ipaddr_status)
+            self.assertEqual(ip_obj.status, ipaddr_status)
+            self.assertFalse(created)
```

Comment:
Not sure I understand this test, especially given that host *is* present in contradiction of the subTest text?